### ISSUE TITLE
Projects using Zarr

### ISSUE THREAD

**Issue by alimanfoo** [zarr-developers/community#19]
If you are using Zarr for whatever purpose, we would love to hear about it. If you can, please leave a couple of sentences in a comment below describing who you are and what you are using Zarr for. This information helps us to get a sense of the different use cases that are important for Zarr users, and helps core developers to justify the time they spend on the project.

**Comment by jrbourbeau** [comment]
Dask array has [`to_zarr`](https://docs.dask.org/en/latest/array-api.html#dask.array.to_zarr)/[`from_zarr`](https://docs.dask.org/en/latest/array-api.html#dask.array.from_zarr) functionality for interfacing with zarr

> helps core developers to justify the time they spend on the project

This is a good point. It could be useful to have a webpage to point to that lists projects using zarr. For example, xarray has an ["Xarray related projects" page](http://xarray.pydata.org/en/stable/related-projects.html) in their sphinx docs and dask has a "Powered by Dask" section in https://dask.org/. Perhaps zarr could have something similar

**Comment by QueensGambit** [comment]
Hello @alimanfoo ,
the **Zarr** library is used to compress the training data for learning a neural network named _CrazyAra_ to play chess variants:
* https://github.com/queensgambit/crazyara

In 2019, **Zarr** is also used for compressing the data which was generated in self-play games in a reinforcement learning setup.
Here, **Zarr** was combined with the [**z5** library](https://github.com/constantinpape/z5) to export data in C++ and subsequently load it in python for training.
I would like to cite this awesome library in my master thesis. Is there a desired way for doing so?
Otherwise I would rely on the default way of citing GitHub repositories:
```bash
@misc{miles2019zarr,
	title = {zarr-developers/zarr-python},
        author = {Miles, Alistair},
	copyright = {MIT},
	url = {https://github.com/zarr-developers/zarr-python},
	abstract = {An implementation of chunked, compressed, N-dimensional arrays for Python.},
	urldate = {2019-12-19},
	publisher = {Zarr Developers},
	month = dec,
	year = {2019},
	note = {original-date: 2015-12-15T14:49:40Z}
}
```

**Comment by tiagoantao** [comment]
I am writing a book for Manning tentatively called "High performance Python for data analytics" and zarr is a big part of it, especially because a good chunk of the book is about persistence efficiency

**Comment by amcnicho** [comment]
Working on a project that is currently exploring the adoption of a stack built around Dask and Zarr (interface via Xarray). This is envisioned as a replacement for a legacy persistent data model underlying the flagging, calibration, synthesis, visualization, and analysis of astronomical data, especially from radio interferometers (e.g., [ALMA](https://almascience.org)).

@alimanfoo, does the Zarr project have a preferred method of citation in academic publications?

**Comment by alimanfoo** [comment]
Just to say thanks everyone for adding projects here, very much appreciated.

Re citation, we don't have a preferred method, [your suggestion](https://github.com/zarr-developers/community/issues/19#issuecomment-567500970) @QueensGambit sounds good in the interim. Writing a short paper about zarr is very much on the wish list.

**Comment by benbovy** [comment]
I've recently added saving simulation data as Zarr datasets in [xarray-simlab](https://github.com/benbovy/xarray-simlab).

https://xarray-simlab.readthedocs.io/en/latest/io_storage.html#using-zarr

Everything went smoothly when working on this. Thanks @alimanfoo and all Zarr developers! (Also thanks @rabernat, @jhamman and others for Zarr integration with Xarray). 

I'm working now on running batches of simulations in parallel using Dask and save all that data in the same Zarr datasets (along a batch dimension), but [I'm struggling with different things](https://github.com/benbovy/xarray-simlab/issues/116) so I might ask for some help.

**Comment by joshmoore** [comment]
https://google.github.io/tensorstore/tensorstore/driver/index.html#chunked-storage-drivers

> A TensorStore is an asynchronous view of a multi-dimensional array. Every TensorStore is backed by a driver, which connects the high-level TensorStore interface to an underlying data storage mechanism. Using an appropriate driver, a TensorStore may be used to access:
>
> * chunked storage formats like zarr, N5, Neuroglancer precomputed, backed by a supported key-value storage system, such as: Google Cloud Storage, Local and network filesystems

**Comment by kindjacket** [comment]
I'm running parallel simulations using dask distributed. I'm using Zarr to create and cache input data to disk for fast reading for each simulation. The data is too large to pass between the parallel processes. I was using Feather before but I needed more complex data structures. Switching to Zarr improved simulation speed by roughly 30%. Thanks for the work on this ❤️

**Comment by gzuidhof** [comment]
Lyft Level 5 just released a dataset in zarr format, it consists of over 1000 hours of driving data and observations. Along with it they have open sourced a codebase relating to the prediction and planning task in autonomous vehicles. Zarr is a great fit for ML problems!

Links:
* [Competition homepage](https://self-driving.lyft.com/level5/prediction)
* [Dataset Whitepaper](https://arxiv.org/abs/2006.14480)
* [Format description](http://www.l5kit.org/data_format.html#lyft-competition-dataset-format)

**Comment by jakirkham** [comment]
Do they happen to have a tweet? May be worth retweeting

**Comment by gzuidhof** [comment]
https://twitter.com/LyftLevel5

**Comment by alimanfoo** [comment]
Thanks @gzuidhof, tweeted from zarr_dev here: https://twitter.com/zarr_dev/status/1277515272270815232

**Comment by jakirkham** [comment]
Thanks both 🙂

**Comment by ericgyounkin** [comment]
Hi, I am building a distributed multibeam sonar processing software suite using dask/xarray/zarr.  Big fan of zarr!  The format is so easy to work with.  Really appreciate the work of this community.

[Kluster](https://github.com/noaa-ocs-hydrography/kluster)

**Comment by PaulJWright** [comment]
We are going to be hosting the Solar Dynamics Observatory (SDO) Machine Learning dataset (https://iopscience.iop.org/article/10.3847/1538-4365/ab1005) on a public-facing Google Cloud bucket, stored in Zarr format! Appreciate the work here, this is fantastic!

**Comment by parashardhapola** [comment]
Hi @alimanfoo and Zarr developers,

First of all, thank you for this developing, maintaining and consistently improving this incredible software.

We have created a memory efficient single-cell genomics data processing toolkit, called **Scarf**, using Zarr.
There is a preprint describing Scarf [here](https://www.biorxiv.org/content/10.1101/2021.05.02.441899v1)
A tweet introducing Scarf can be found [here](https://twitter.com/GKarlssonLab/status/1389560499214426117)

We chose Zarr over HDF5 because:
- support for highly performant compression libraries like LZ4
- parallel read support through Dask.
- random access to data supported on AWS S3
- rectangular chunking that allows data to be loaded in batches of both rows and column

**Comment by RichardScottOZ** [comment]
Using it to enable country scale machine learning geology modelling from Sentinel data.

**Comment by jhamman** [comment]
We're using Zarr for a new visualization application, rending chunks of data directly from cloud object store using WebGL and Mapbox. A full writeup of our approach is here: https://carbonplan.org/blog/maps-library-release

**Comment by aladinor** [comment]
Hi there! I am using Zarr to convert several hdf5 files coming from NASA P3 aircraft (planes that fly inside clouds including hurricanes). This conversion allows us to manage/access large data set for machine learning applied to cloud microphysics.

**Comment by magnunor** [comment]
Hey, we've added support for `zarr` in the HyperSpy library, which is a library for processing electron microscopy data. These last years, we've been getting faster and faster detectors, so our datasets have drastically increased in size: from 100 MBs to 100+ GBs. Previously we used `HDF5` for this, but using `zarr` has greatly improved performance for working with these large datasets.

Kudos to @cssfrancis for adding this to HyperSpy: https://github.com/hyperspy/hyperspy/pull/2825

https://hyperspy.org/hyperspy-doc/current/user_guide/io.html#zspy-format

**Comment by julioasotodv** [comment]
Hi! At Sandoz Pharmaceuticals we use `zarr` to store large, compressed n-dimensional arrays for a wide variety of projects, including biosimilars structure optimization with high-dimensional (>1M-dimensional) bayesian optimization, storage of large adjacency matrices for large molecular graphs (some of these matrices are of dimensions 30M x 30M), and for medical imaging data.

Multi-threaded reads and writes using Dask have saved us hours of waiting!

Unfortunately I can't share any further details due to trade secrets, but I am really thankful for your great work with this storage format 😄

**Comment by miccoli** [comment]
Hi, zarr has become a cornerstone of all my data acquisition pipelines. The main “selling point“ for me is how easy it is to build very robust systems, with limited data loss in case of hardware/software failures.

**Comment by PowerChell** [comment]
Hello everyone, I am a Developer Advocate from [Radiant Earth](https://radiant.earth/), and I love seeing all of this Zarr work. If any of you are interested in hosting your Zarr data on [Source Cooperative](https://source.coop/) (currently in public beta at [beta.source.coop](https://beta.source.coop/)), please email me at [hello@source.coop](mailto:hello@source.coop). We would love to increase exposure to any Zarr projects that want to be seen/shared. 

Also interested in hosting other cloud-optimized data formats on Source as well. To read more about Source, please see [this blog post](https://radiant.earth/blog/2023/10/what-is-source-cooperative/).

**Comment by jeromekelleher** [comment]
I've used Zarr to store large-scale genetic variation data efficiently in [tsinfer](https://www.nature.com/articles/s41588-019-0483-y)

I'm also involved in the [sgkit project](https://pystatgen.github.io/sgkit/latest/) which uses Zarr to work with genomics data more generally (and currently writing a paper about it). Zarr is awesome :+1:
